agent/xds/routes.go

```diff
@@ -599,44 +611,43 @@ func makeUpstreamRouteForDiscoveryChain(
 		}
 
 	case structs.DiscoveryGraphNodeTypeSplitter:
-		routeAction, err := makeRouteActionForSplitter(startNode.Splits, chain, clusterNamePrefix)
-		if err != nil {
-			return nil, err
-		}
+		routeAction, err := s.makeRouteActionForSplitter(upstreamsSnapshot, startNode.Splits, chain, forMeshGateway)
```

Comment:
Why not return if there's an error here? Also the error isn't being checked anymore